tests/upgrades/conftest.py

```diff
@@ -223,7 +231,17 @@ def test_something_post_upgrade(pre_upgrade_data):
     data = [_read_test_data(test_node_id) for test_node_id in depend_on_node_ids]
     if len(data) == 1:
         data = data[0]
-    return data
+    return Box(data)
+
+
+@pytest.fixture(scope='class')
+def class_pre_upgrade_data(request):
+    data = {
+        key: value
+        for key, value in get_all_entity_data().items()
+        if key.startswith(f"tests/upgrades/{request.node.parent.name}::{request.node.name}")
```

Comment:
Please use relative path since upgrade scenarios can be run from `tests/upgrades` directory and it might fail here.